### ISSUE TITLE
How to detect data casting

### ISSUE THREAD

**Issue by chuayupeng** [facebook/mariana-trench#57]
Is it possible for mariana trench to detect if the data is being casted into (Intent) for example? How would the source definition look like in that case?

**Comment by arthaud** [comment]
Could you give another example? `(String)` is not valid on the left hand side of an assignment.

**Comment by chuayupeng** [comment]
Hi @arthaud,

I want to differentiate between these 2 lines of code, so I can reduce false positives when detecting intent redirections. Is there any way to detect what the data is casted into?

```
Intent extraIntent = (Intent) intent.getExtras("...");
```
and 
```
String extraString = (String) intent.getExtras("...");
```

**Comment by arthaud** [comment]
Apparently, this generates a `CHECK_CAST v1, Ljava/lang/String;` in Dex. We currently ignore those, so there is no way to differentiate these, unfortunately.

**Comment by arthaud** [comment]
That should be solved with a87023dc2ffac0de3bf00f864d9da6b269ad0775
